control-plane/typeSpecConverter.cpp

```diff
@@ -111,6 +115,52 @@ bool TypeSpecConverter::preorder(const IR::Type_Name* type) {
     return false;
 }
 
+bool TypeSpecConverter::preorder(const IR::Type_Newtype* type) {
+    if (p4RtTypeInfo) {
+        bool orig = true;
+        cstring uri;
+        uint32_t sdnB;
+        auto ann = type->getAnnotation("p4runtime_translation");
+        if (ann != nullptr) {
+            if (ann->expr.size() > 1) {
+                ::error("invalid annotation %1%", ann->expr);
+            } else {
+                orig = false;
+                for (auto a : ann->body) {
+                    if (a->text.startsWith("p4.org/psa/v1")) {
```

Comment:
This is not correct, this code is used by many architectures, not just PSA and `@p4runtime_translation` is not limited to PSA, which means you cannot identify the URI like this.
I also don't know why this is required, you *know* that the first argument is the URI and the second is the bitwidth...

Reply:
I changed the code to use an `int i` variable in the for loop and if `i==0` the first arg is parsed.

Reply:
> This is not correct, this code is used by many architectures, not just PSA and `@p4runtime_translation` is not limited to PSA, which means you cannot identify the URI like this.
> I also don't know why this is required, you _know_ that the first argument is the URI and the second is the bitwidth...

To be precise, there are three `AnnotationToken(s)`.  The third is a "comma" between the URI and the  sdn_bitwidth.